ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraphArrow/utils.ts

```diff
@@ -26,16 +32,20 @@ export function nodeLabel(
   showBinaryName: boolean
 ): string {
   const functionName: string | null = table.getChild(FIELD_FUNCTION_NAME)?.get(row);
+  const labelsOnly: boolean | null = table.getChild(FIELD_LABELS_ONLY)?.get(row);
+  const labels: string | null = table.getChild(FIELD_LABELS)?.get(row);
+  console.log(labelsOnly, labels);
   if (functionName !== null && functionName !== '') {
-    if (level === 1 && functionName.startsWith('{') && functionName.endsWith('}')) {
-      return Object.entries(JSON.parse(functionName))
-        .sort(([a], [b]) => a.localeCompare(b))
-        .map(([k, v]) => `${k}="${v as string}"`)
-        .join(', ');
-    }
     return functionName;
   }
 
+  if (level === 1 && labelsOnly !== null && labelsOnly && labels !== null && labels !== '') {
```

Comment:
If we now have this `labelsOnly` column to read from, we can probably remove passing the `level` all the way down.